main.js

```diff
@@ -0,0 +1,156 @@
+        imageUrl: 'https://goo.gl/Cro3tL'
+      },
+      {
+        itemId: 7,
+        name: 'Bodhran',
+        brand: 'Meinl',
+        price: 129.99,
+        description: 'A frame drum for people who like to jig.',
+        details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+        origin: 'Ireland',
+        imageUrl: 'https://goo.gl/YNb74O'
+      },
+      {
+        itemId: 8,
+        name: 'TR-09',
+        brand: 'Roland',
+        price: 399,
+        description: 'A drum machine for people with no desk space.',
+        details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
+        origin: 'Japan',
+        imageUrl: 'https://goo.gl/gJYUxz'
+      }
+    ]
+  },
+  details: {
+    item: null
+  }
+}
+
+var itemsPerCol = 3
+
+function renderItem(item) {
+  var $item =
+    createElement('div', { class: 'card border border-dark border-3', style: 'width: 18rem height:28rem' }, [
+      createElement('div', { style: 'height: 320px' }, [
+        createElement('img', { class: 'card-img-top', src: item.imageUrl, alt: 'Card image cap' }, [])
+      ]),
+      createElement('div', { style: 'height: 280px' }, [
+        createElement('div', { class: 'card-body p-1', style: 'height: 180px;' }, [
+          createElement('h3', { class: 'card-title' }, [item.name]),
+          createElement('div', { class: 'align-text-bottom' }, [
+            createElement('h5', { class: 'card-text' }, [item.brand]),
+            createElement('p', { class: 'card-text' }, ['$' + item.price])
+          ])
+        ]),
+        createElement('div', { style: 'footer' }, [
+          createElement('button', { class: 'btn bg-primary btn-lg btn-block p-3', style: 'position: absolute; height: 100px; border: solid; bottom: 0; right: 0;  left: 0', id: 'add-to-cart' }, ['Add to Cart'])
+        ])
+      ])
+    ])
+  return $item
+}
+
+function renderGrid(gridElements) {
+  var $header = createElement('h1', { class: 'text-center mx-auto text-primary m-3 display-1', style: 'width: 500px' }, ['JAMAZON'])
+  var $container = createElement('div', { class: 'container' }, [])
+  $container.appendChild($header)
+  for (var j = 0; j < gridElements.length / itemsPerCol; j++) {
```

Comment:
Just cram all of the columns into one row. Bootstrap knows when to stack them, just choose the right variant of the `.col-` class. You can add as many as you want to make sure the columns stack at the right viewport sizes.